src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -289,10 +289,23 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
         return this.isShowingMore ? showMoreLimit : limit;
       },
 
-      getConfiguration: (configuration = {}) => {
-        const widgetConfiguration = {
-          [operator === 'and' ? 'facets' : 'disjunctiveFacets']: [attribute],
-        };
+      getConfiguration(configuration) {
+        let widgetConfiguration;
+        if (operator === 'and') {
+          widgetConfiguration = {
+            facets: [attribute],
```

Comment:
https://github.com/algolia/instantsearch.js/pull/3986#discussion_r307677462

Reply:
How do we want to deal with facets duplication here? Call `uniq` at this step?

Reply:
Actually I thought that the `SearchParameters` managed that for us but I was wrong it only does it for the facets refinements. Whether the class does it for us or not the behaviour should be consistent. Keep it like this then the merge function will do it.

Once we merge the two lifecycles (`getConfiguration` & `getWidgetSearchParameters`) we should be able get rid of the previous value since we'll always clear the previous value to let the `uiState` completely control the refinements.

Reply:
Actually I was correct. `SearchParameters` manage it for us, at least for [`facets`](https://github.com/algolia/algoliasearch-helper-js/blob/6ac260dfcbdc0a94059eb93b8ad8b4552f63abf5/src/SearchParameters/index.js#L689-L697), [`disjunctiveFacets`](https://github.com/algolia/algoliasearch-helper-js/blob/6ac260dfcbdc0a94059eb93b8ad8b4552f63abf5/src/SearchParameters/index.js#L705-L713) but not for [`hierarchicalFacets`](https://github.com/algolia/algoliasearch-helper-js/blob/6ac260dfcbdc0a94059eb93b8ad8b4552f63abf5/src/SearchParameters/index.js#L722-L731). The deduplication for facet refinements are also managed by the `SearchParameters` for each type of facets. 

cc @Haroenv @francoischalifour